packages/grid/x-data-grid-pro/src/hooks/features/export/gridExcelExportInterface.ts

```diff
@@ -0,0 +1,61 @@
+import type * as Excel from 'exceljs';
+import {
+  GridFileExportOptions,
+  GridExportFormat as GridExportFormatCommunity,
+  GridExportExtension as GridExportExtensionCommunity,
+  GridToolbarExportProps as GridToolbarExportPropsCommunity,
+  GridExportDisplayOptions,
+} from '@mui/x-data-grid';
+
+export type GridExportFormat = GridExportFormatCommunity | 'excel';
+export type GridExportExtension = GridExportExtensionCommunity | 'xlsx';
+
+export interface GridExceljsProcessInput {
+  workbook: Excel.Workbook;
+  worksheet: Excel.Worksheet;
+}
+
+/**
+ * The options to apply on the Excel export.
+ */
+export interface GridExcelExportOptions extends GridFileExportOptions {
+  /**
+   * Name given to the worksheet containing the columns valueOptions.
+   * valueOptions are added to this worksheet if they are provided as an array.
+   */
+  valueOptionsSheetName?: string;
+  /**
+   * Method applied before adding the grid content to the workbook
+   * @param {GridExceljsProcessInput} processInput object containing the workbook and the worksheet
+   * @returns {Promise<void>} Modifications are done by mutating the workbook or the worksheet
+   * */
+  exceljsPreProcess?: (processInput: GridExceljsProcessInput) => Promise<void>;
+  /**
+   * Method applied after adding the grid content to the workbook
+   * @param {GridExceljsProcessInput} processInput object containing the workbook and the worksheet
+   * @returns {Promise<void>} Modifications are done by mutating the workbook or the worksheet
+   * */
+  exceljsPostProcess?: (processInput: GridExceljsProcessInput) => Promise<void>;
+}
+
+export interface GridToolbarExportProps extends GridToolbarExportPropsCommunity {
+  excelOptions: GridExcelExportOptions & GridExportDisplayOptions;
+}
+
+/**
+ * The excel export API interface that is available in the grid [[apiRef]].
+ */
+export interface GridExcelExportApi {
+  /**
+   * Returns the grid data as an exceljs workbook.
+   * This method is used internally by `exportDataAsExcel`.
+   * @param {GridExcelExportOptions} options The options to apply on the export.
+   * @returns {Promise<Excel.Workbook>} The data in a exceljs workbook object.
+   */
+  getDataAsExcel: (options?: GridExcelExportOptions) => Promise<Excel.Workbook> | null;
+  /**
+   * Downloads and exports an Excel file of the grid's data.
+   * @param {GridExcelExportOptions} options The options to apply on the export.
```

Comment:
It also returns a promise.

```suggestion
   * @param {GridExcelExportOptions} options The options to apply on the export.
   * @returns {Promise<void>} A promise which resolves after exporting to Excel.
```